pkgs/apps/dirac/default.nix

```diff
@@ -0,0 +1,131 @@
+{ stdenv, lib, fetchurl, gfortran, cmake, makeWrapper
+, which, openssh, blas, lapack, mpi, boost, exatensor, python3
+} :
+
+assert
+  lib.asserts.assertMsg
+  (!blas.isILP64)
+  "A 32 bit integer implementation of BLAS is required.";
+
+stdenv.mkDerivation rec {
+  pname = "dirac";
+  version = "21.0";
+
+  nativeBuildInputs = [
+    which
+    gfortran
+    cmake
+    python3
+    makeWrapper
+    openssh
+  ];
+
+  # For now, DIRAC is built without the PCMSolver optional dependency.
+  buildInputs = [
+    blas
+    lapack
+    boost
+    exatensor
+  ];
+
+  propagatedBuildInputs = [
+    mpi
+  ];
+
+  src = fetchurl {
+    url = "https://zenodo.org/record/4836496/files/DIRAC-${version}-Source.tar.gz";
+    sha256 = "0vqmin24xqkcp3rvml0dimc4kjmb3m37c3l6bkjy4y6d3xyb41yg";
+  };
+
+  prePatch = ''
+    ls -lah
+  '';
+
+  patches = [
+    ./0001-exatensor-cmake.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace ./cmake/custom/exatensor.cmake \
+    --subst-var-by exatensor ${exatensor}
+
+    patchShebangs .
+  '';
+
+  FC = "mpif90";
+  CC = "mpicc";
+  CXX = "mpicxx";
+  MATH_ROOT = blas;
+  ENABLE_EXATENSOR = "ON";
+
+  /*
+  Cmake is required to build but adding it to the buildinputs then ignores the
+  setup script. Therefore i call the script here manually but cmake is invoked
+  by setup.
+  */
+
+  configurePhase = ''
+    ./setup --prefix=$out --mpi
+  '';
+
+  preBuild = ''
+    cd build
+  '';
+
+  enableParallelBuilding = true;
+  hardeningDisable = [ "format" ];
+  doInstallCheck = true;
+
+  /*
+  Make the MPI stuff available to the DIRAC script by hard-coding the MPI path.
+  Moving pam-dirac to pam to conform with the DIRAC manual. Usually there is a
+  weird hack in the install.cmake to install pam into the bin folder.
+
+  The pam script is just a Python script that sets a ton of ENV variables.
+  */
+  postFixup = ''
+    mv $out/bin/pam-dirac $out/bin/pam
```

Comment:
Should the name of the target not be `pam-dirac`? Is there a reason it gets modified/shortened?

Reply:
Yes, the documentation of DIRAC always refers to the script as `pam`. Internally in the CMake process it copies the file called `pam` to `pam-dirac` and then does some weird trick to hack CMake installation. But in the end, it wants to have a main entrypoint called `pam`.